test/support/utils_err_eng.h

```diff
@@ -0,0 +1,188 @@
+         *      ready to receive from Kernel part of test. All additional errors
+         *      will be ignored.
+         */
+        ErrorContainer_HostPart(::std::size_t _max_errors_count)
+            : errors(_max_errors_count)
+        {
+        }
+
+        static cl::sycl::buffer<ErrorInfo, 1> get_sycl_buffer(ErrorContainer_HostPart& host_part)
+        {
+            const cl::sycl::range<1> numOfItems{host_part.errors.size()};
+            return cl::sycl::buffer<ErrorInfo, 1>(host_part.errors.data(), numOfItems);
+        }
+
+        bool have_errors() const
+        {
+            return errors[0].isError;
+        }
+
+        /// Process errors, occurred in Kernel, end exit if has errors
+        void process_errors()
+        {
+            for (const auto& error : errors)
+            {
+                if (!error.isError)
+                    break;
+
+                expect(true, false,    // true != false -> is error
+                       error.file,
+                       error.line,
+                       error.msg,
+                       false);         // Do not exit on error
+            }
+        }
+    };
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// struct ErrorContairer_KernelPart - kernel part of error container for kernel tests
+    template <typename TAccessor>
+    struct ErrorContairer_KernelPart
+    {
+        const ::std::size_t max_errors_count = 0;
+        TAccessor error_buf_accessor;
+        unsigned index = 0;
+
+        ErrorContairer_KernelPart(TAccessor acc, ::std::size_t _max_errors_count)
+            : error_buf_accessor(acc), max_errors_count(_max_errors_count)
+        {
+        }
+
+        template <typename T>
+        std::size_t get_buf_size(const T& buf) const
+        {
+            return sizeof(buf) / sizeof(buf[0]);
+        }
+
+        void expect_true(bool condition, const char* file, std::int32_t line, const char* msg)
+        {
+            if (!condition)
+                add_error_info(file, line, msg);
+        }
+
+        void add_error_info(const char* file, std::int32_t line, const char* msg)
+        {
+            assert(index < max_errors_count);
+            if (index == max_errors_count)
+                return;
+
+            auto& ei = error_buf_accessor[index];
+            copy_string(ei.file, get_buf_size(ei.file), file);
+            ei.line = line;
+            copy_string(ei.msg, get_buf_size(ei.msg), msg);
+            copy_string(ei.msg + get_str_len(ei.msg), get_buf_size(ei.msg) - get_str_len(ei.msg), " (in Kernel)");
+            ei.isError = true;
+
+            ++index;
+        }
+
+        ::std::size_t get_str_len(const char* buf) const
+        {
+            ::std::size_t len = 0;
+
+            for (; *buf != 0x00; ++buf)
+                ++len;
+
+            return len;
+        }
+
+        void copy_string(char* dest, std::size_t dest_size, const char* src) const
```

Comment:
Actually, I dont understand why we need any function for string processing here... I would just  remind - we ae testing std::complex here, not std::string etc)

Reply:
@MikeDvorskiy What do you suggest to use instead in the lines 138-141 above?

Reply:
I would suggest remove it.... To tell the truth, I think we don't need string processing at all. I suggest a very simple design/approach: using static asserts for compile time checks and RT asserts for RT errors, no any string processing in kernel (because we dont need it.